dwh/query_lib_test.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import query_lib
+
+
+class PatientQueryTest(unittest.TestCase):
+
+  def test_single_code_with_values(self):
+    patient_query = query_lib.PatientQuery()
+    patient_query.include_obs_values_in_time_range(
+        'TEST_CODE', ['VAL1', 'VAL2'], '2021-06-01', '2021-07-10')
+    sql_constraint = patient_query.all_constraints_sql()
+    self.assertEqual(sql_constraint, (
+        '((dateTime >= "2021-06-01" AND dateTime <= "2021-07-10" AND '
+        'coding.code="TEST_CODE" AND value.string IN ("VAL1","VAL2")))'))
+
+  def test_two_codes_with_values_and_range(self):
+    patient_query = query_lib.PatientQuery()
+    patient_query.include_obs_values_in_time_range(
+        'TEST_CODE1', ['VAL1', 'VAL2'], '2021-06-01', '2021-07-10')
+    patient_query.include_obs_in_value_and_time_range(
+        'TEST_CODE2', 0.1, None, '2021-07-09', '2021-07-10')
+    sql_constraint = patient_query.all_constraints_sql()
+    self.assertEqual(sql_constraint, (
+        '((dateTime >= "2021-06-01" AND dateTime <= "2021-07-10" AND '
```

Comment:
Quick question? Let's say I'm interested in knowing how many patients are currently on ART. I pass in my min and max dates as "2021-06-01" and  "2021-07-10" respectively. If a patient had his/her last observation in `2021-04` will  the query-lib include him/her in the filter?

Reply:
No, if you need them to be counted then you should pass the longer period, at least for that particular code that determines ART (for others a shorter period can be used). Does this make sense?

Reply:
Agreed, we will need to adjust for this scenario -->  So at any given point a patient can be given an RCT date ranging between 1 month to 3 months, depending on how the patient is doing. For example, a stable patient who is suppressed can be given an RCT date of up to 3 to 6 months.  Does that mean for such indicators, we will need to pass longer max dates of up to 6 months? If so, how will filter out patients who are truly not on ART, ie., they had `ART_PLAN=CONTINUE_Regimen` at month 1 and never showed up? The above constraint will count him/her as on ART, and please correct me if that's not the case.

Reply:
this should be resolved by the new commit, thanks for working on this